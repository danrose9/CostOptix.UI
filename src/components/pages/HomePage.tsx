import React from 'react';
import {
  StyledHeader,
  StyledSubHeader,
  PageContainer,
  HomePageMainContent,
  HomePageMainLeft,
  HomePageTitle,
  HomePageSubTitle,
  HomePageActions,
  HomePageButton,
} from '../__styles__/HomePageStyles';
import { BASE, DEMO_LOGIN } from '../../services/api/apiEndpoints';
import HomePageNav from '../navbar/HomePageNav';
import { AppDispatch } from 'src/services/redux/store';
import { useDispatch } from 'react-redux';
import { setIsDemo } from '../../services/redux/reducers/userProfileSlice';

export const HomePage = () => {
  const dispatch = useDispatch();

  const handleDemoClick = () => {
    dispatch<AppDispatch>(setIsDemo(true));
  };

  return (
    <>
      <PageContainer fluid data-testid="homePage-2" className="home-page">
        <HomePageNav />

        <HomePageMainContent>
          <HomePageMainLeft>
            <HomePageTitle>
              <StyledHeader>Stop Over-Spending on Cloud Services</StyledHeader>
            </HomePageTitle>
            <HomePageSubTitle>
              <StyledSubHeader>
                A powerful and insightful software suite to explore, understand and take action on cloud services
                spending
              </StyledSubHeader>
            </HomePageSubTitle>
            <HomePageActions>
              <form method="POST" action={BASE + DEMO_LOGIN}>
                <HomePageButton className="view-demo" data-testid="login-demo-button" onClick={handleDemoClick}>
                  View Demo
                </HomePageButton>
              </form>
            </HomePageActions>
          </HomePageMainLeft>
        </HomePageMainContent>
      </PageContainer>
    </>
  );
};

export default HomePage;
