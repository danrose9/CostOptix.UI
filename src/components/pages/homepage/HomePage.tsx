import React from 'react';
import {
  StyledHeader,
  StyledSubHeader,
  PageContainer,
  HomePageMainContent,
  HomePageMainLeft,
  HomePageTitle,
  HomePageSubTitle,
  HomePageActions,
  HomePageButton,
} from '../../__styles__/HomePageStyles';
import { BASE, DEMO_LOGIN } from '../../../services/api/apiEndpoints';
import HomePageNav from './HomePageNav';
import { AppDispatch } from 'src/services/redux/store';
import { useDispatch } from 'react-redux';
import { setIsDemo } from '../../../services/redux/reducers/userProfileSlice';
import CookieBanner from '../../CookieBanner';
import ReactGA from 'react-ga4';
import { eventTypes } from '../../../hooks/useTrackEvent';
import MessageBox from '../../MessageBox';
import { useMobileDevice } from 'src/hooks/useMobileDevice';
import { Container } from 'semantic-ui-react';

const content = {
  TITLE: 'Reduce Your Cloud Spending Costs Effectively with CostOptix',
  SUBTITLE:
    'A powerful and insightful software suite to explore, understand and take action on cloud services spending.',
  DEMO_BUTTON: 'View Demo',
};

const mobile = {
  TITLE: 'CostOptix is now available!',
  MESSAGE: 'For the best experience, please use a desktop or tablet device.',
};

const HomePage = () => {
  const dispatch = useDispatch();

  const handleDemoClick = () => {
    ReactGA.event(eventTypes.DEMO_LOGIN, { user: 'demo' });
    dispatch<AppDispatch>(setIsDemo(true));
  };

  const MobileMessageBox = useMobileDevice(
    <Container>
      <MessageBox title={mobile.TITLE} message={mobile.MESSAGE} color="yellow" />
    </Container>,
    640
  );

  const LoginDemoButton = useMobileDevice(
    <HomePageActions>
      <form method="POST" action={BASE + DEMO_LOGIN}>
        <HomePageButton className="view-demo" data-testid="login-demo-button" onClick={handleDemoClick}>
          {content.DEMO_BUTTON}
        </HomePageButton>
      </form>
    </HomePageActions>,
    640,
    false
  );

  return (
    <>
      <PageContainer fluid data-testid="homePage-2" className="home-page">
        <HomePageNav />

        <HomePageMainContent>
          <HomePageMainLeft>
            <HomePageTitle>
              <StyledHeader>{content.TITLE}</StyledHeader>
            </HomePageTitle>
            {MobileMessageBox}
            <HomePageSubTitle>
              <StyledSubHeader>{content.SUBTITLE}</StyledSubHeader>
            </HomePageSubTitle>
            {LoginDemoButton}
          </HomePageMainLeft>
          <CookieBanner />
        </HomePageMainContent>
      </PageContainer>
    </>
  );
};

export default HomePage;
