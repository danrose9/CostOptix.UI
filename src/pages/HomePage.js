import React from 'react';
import {
  StyledHeader,
  StyledSubHeader,
  PageContainer,
  HomePageNav,
  HomePageMainContent,
  HomePageFooter,
  HomePageMainLeft,
  HomePageTitle,
  HomePageSubTitle,
  HomePageActions,
} from './__styles__/HomePageStyles';
import { GetStartedModal } from '../components/modals/GetStartedModal';
import { Login, LoginDemo } from '../components/auth/Login';
import { Image, Menu } from 'semantic-ui-react';
import * as images from '../assets/index';

export const HomePage = () => {
  return (
    <>
      <PageContainer fluid data-testid="homePage-2">
        <HomePageNav>
          <Menu fluid secondary color="green">
            <Menu.Item>
              <div>
                <Image src={images.LOGOBLUE} size="medium" />
              </div>
            </Menu.Item>
            <Menu.Menu position="right">
              {/* <MenuItem name="Contact" />
              <MenuItem name="Pricing" />
              <MenuItem name="About Us" />
              <MenuItem name="Legal" /> */}
              <Menu.Item>
                <Login disabled={false} primary={false} content="Login" size="big" color="black" />
              </Menu.Item>
            </Menu.Menu>
          </Menu>
        </HomePageNav>
        <HomePageMainContent>
          <HomePageMainLeft>
            <HomePageTitle>
              <StyledHeader>Stop Over-Spending on Cloud Services</StyledHeader>
            </HomePageTitle>
            <HomePageSubTitle>
              <StyledSubHeader>
                A powerful and insightful software suite to explore, understand and take action on cloud services
                spending
              </StyledSubHeader>
            </HomePageSubTitle>
            <HomePageActions>
              <GetStartedModal action="Get Started" />
              <LoginDemo
                style={{ fontSize: '1.7em' }}
                disabled={false}
                primary={true}
                size="huge"
                content="View Demo"
              />
            </HomePageActions>
          </HomePageMainLeft>
        </HomePageMainContent>
        <HomePageFooter></HomePageFooter>
      </PageContainer>
    </>
  );
};

export default HomePage;
