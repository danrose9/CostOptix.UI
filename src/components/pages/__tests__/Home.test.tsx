import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import HomePage from '../HomePage';
import { BrowserRouter as Router } from 'react-router-dom';
import { ApplicationWrapper } from '../../../tests/helpers';
import renderer from 'react-test-renderer';

afterEach(() => {
  cleanup();
});

describe('Home Page', () => {
  test('should render Home Page', () => {
    render(<HomePage />, { wrapper: ApplicationWrapper });
  });

  test('should contain buttons', () => {
    const { getByTestId } = render(<HomePage />, { wrapper: ApplicationWrapper });

    expect(getByTestId('login-ext-button')).toBeEnabled();
    expect(getByTestId('login-demo-button')).toBeEnabled();
    expect(getByTestId('get-started-button')).toBeEnabled();
  });

  test('matches snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <HomePage />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
