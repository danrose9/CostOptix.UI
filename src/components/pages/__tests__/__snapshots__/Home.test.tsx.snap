// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches snapshot 1`] = `
<div
  className="ui fluid container sc-ftTHYK cHWSdq home-page"
  data-testid="homePage-2"
>
  <div
    className="sc-hLBbgP dFDHev"
  >
    <div
      className="ui green fluid secondary menu"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        <div
          className="sc-eDvSVe gTUNTl"
        >
          <img
            className="ui medium image"
            onClick={[Function]}
            src="logo-blue.png"
          />
        </div>
      </div>
      <div
        className="right menu"
      >
        <button
          className="sc-hBxehG bViUOI transparent shift-right"
          data-testid="support-button"
          onClick={[Function]}
          role="button"
        >
          Help Centre
        </button>
        <div
          aria-expanded={false}
          className="ui simple dropdown sc-bcXHqe kzsyIS"
          data-testid="dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-atomic={true}
            aria-live="polite"
            className="divider text"
            role="alert"
          >
            Legal
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition sc-gswNZR eulPVq"
          >
            <div
              className="item sc-dkrFOg iWHbYH"
              onClick={[Function]}
              role="option"
            >
              Privacy
            </div>
            <div
              className="item sc-dkrFOg iWHbYH"
              onClick={[Function]}
              role="option"
            >
              Terms of Service
            </div>
          </div>
        </div>
        <button
          className="sc-hBxehG bViUOI login"
          data-testid="login-ext-button"
          onClick={[Function]}
          role="button"
        >
          Login
        </button>
        <button
          className="sc-hBxehG bViUOI get-started"
          data-testid="get-started-button"
          onClick={[Function]}
          role="button"
        >
          Get Started
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-pyfCe kKPFa"
  >
    <div
      className="sc-ipEyDJ eaZUbE"
    >
      <div
        className="sc-csuSiG hrYaiu"
      >
        <h1
          className="sc-jrcTuL hjSAun"
        >
          Reduce Your Cloud Spending Costs Effectively with CostOptix
        </h1>
      </div>
      <div
        className="sc-eDWCr iWmlnB"
      >
        <p
          className="sc-kDvujY UufDO"
        >
          A powerful and insightful software suite to explore, understand and take action on cloud services spending
        </p>
      </div>
      <div
        className="sc-bqWxrE iqHtSB"
      >
        <form
          action="undefinedAccount/DemoLogin?returnUrl=/init"
          method="POST"
        >
          <button
            className="sc-hBxehG bViUOI view-demo"
            data-testid="login-demo-button"
            onClick={[Function]}
          >
            View Demo
          </button>
        </form>
      </div>
    </div>
    <div
      className="CookieConsent"
      style={
        Object {
          "alignItems": "baseline",
          "background": "#353535",
          "bottom": "0",
          "color": "white",
          "display": "flex",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "left": "0",
          "position": "fixed",
          "width": "100%",
          "zIndex": "999",
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "flex": "1 0 300px",
            "margin": "15px",
          }
        }
      >
        This site uses cookies.
      </div>
      <div
        className=""
      >
        <button
          aria-label="Accept cookies"
          className=""
          id="rcc-confirm-button"
          onClick={[Function]}
          style={
            Object {
              "background": "#ffd42d",
              "border": "0",
              "borderRadius": "0px",
              "boxShadow": "none",
              "color": "black",
              "cursor": "pointer",
              "flex": "0 0 auto",
              "margin": "15px",
              "padding": "5px 10px",
            }
          }
        >
          I understand
        </button>
      </div>
    </div>
  </div>
</div>
`;
